test/browser/getDomSibling.test.js

```diff
@@ -0,0 +1,353 @@
+import { h, render, Fragment } from '../../src/index';
+import { getDomSibling } from '../../src/component';
+import { setupScratch, teardown } from '../_util/helpers';
+
+/** @jsx h */
+
+describe('getDomSibling', () => {
```

Comment:
@marvinhagemeister FYI, I copied your tests from #1605 and added a bunch. I couldn't get the implementation of `getDomSibling` in #1605 to pass all the new tests which is why I re-built it from scratch.

Reply:
I'm glad you did that as my version was a bit hacked together to get a good sense of the problem space, but I didn't have time to clean it up. The variant in this PR is much cleaner and the `childIndex` handling is ace :100: